lib/js/components/Modals/Modal/Modal.vue

```diff
@@ -0,0 +1,417 @@
+						</wnl-button>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../../styles/settings/spacings';
+@import '../../../../styles/settings/radiuses';
+@import '../../../../styles/settings/shadows';
+@import '../../../../styles/settings/colors/tokens';
+@import '../../../../styles/settings/typography/tokens';
+@import '../../../../styles/settings/icons';
+@import '../../../../styles/settings/media-queries';
+
+$modal-medium-width: 700px;
+$modal-small-width: 460px;
+$image-height: 200px;
+$image-height-small: 140px;
+
+.modal {
+	$self: &;
+
+	background: $color-default-overlay;
+	padding: $space-l $space-s;
+
+	@media #{breakpoint-s()} {
+		padding: $space-l;
+	}
+
+	&__wrapper {
+		background: $color-default-background;
+		border-radius: $radius-m;
+		box-shadow: $shadow-xl;
+		margin: 0 auto;
+		overflow: hidden;
+		position: relative;
+		max-width: $modal-medium-width;
+
+		&.-small {
+			max-width: $modal-small-width;
+
+			#{$self}__image {
+				height: $image-height-small;
+			}
+			#{$self}__content {
+				padding-left: $space-m;
+				padding-right: $space-m;
+			}
+		}
+	}
+
+	&__content {
+		padding: $space-l $space-s;
+
+		@media #{breakpoint-s()} {
+			padding: $space-l $space-xl;
+		}
+
+		&.-centered {
+			#{$self}__header,
+			#{$self}__textContent {
+				text-align: center;
+			}
+		}
+	}
+
+	&__header {
+		display: flex;
+		flex-direction: column;
+	}
+
+	&__headerFeatureIcon {
+		align-self: center;
+		margin-bottom: $space-m;
+	}
+
+	&__headerTitle {
+		@include displayHeading-xs-default-bold;
+
+		margin-bottom: $space-xs;
+		margin-top: 0;
+
+		&.-small {
+			@include heading-xl-default-bold;
+		}
+	}
+
+	&__headerSubtitle {
+		@include heading-m-default-regular;
+
+		margin-bottom: $space-s;
+		margin-top: 0;
+	}
+
+	&__textContent {
+		@include text-m-default-regular;
+
+		margin-bottom: $space-s;
+	}
+
+	&__checkbox {
+		cursor: pointer;
+		display: flex;
+		align-items: center;
+	}
+
+	&__checkboxLabel {
+		color: $color-neutral-text-heavy;
+		cursor: pointer;
+		margin-left: $space-xxs;
+
+		/** align vertical middle fix */
+		margin-top: -$space-xxxxxs;
+	}
+
+	&__close {
+		border-radius: 100%;
+		color: $color-neutral-icon-weak;
+		position: absolute;
+		right: $space-xs;
+		top: $space-xs;
+	}
+
+	&__image {
+		height: $image-height;
+		background-size: cover;
+	}
+
+	&__footer {
+		display: flex;
+		justify-content: space-between;
+		flex-direction: column-reverse;
+		padding-top: $space-m;
+
+		@media #{breakpoint-s()} {
+			flex-direction: row;
+		}
+
+		&.-singleColumn {
+			#{$self}__footerColumn {
+				width: 100%;
+				justify-content: center;
+			}
+		}
+	}
+
+	&__footerColumn {
+		display: flex;
+		gap: 0 $space-s;
+
+		&.--cta {
+			flex-direction: column-reverse;
+			margin-bottom: $space-xs;
+
+			@media #{breakpoint-s()} {
+				flex-direction: row;
+				margin-bottom: 0;
+			}
+		}
+
+		&.--ctaSecondary {
```

Comment:
if we have checkbox and button, should they be rendered in the same line on mobile?

Reply:
I asked UX about it. There was mistake in Figma, the shouldn't be in the same line on the mobile. Btw. this situation shouldn't happen.

Reply:
it isn't clear for me. They shouldn't be on the same line or this situation shouldn't happen? If the first statement is correct, then we have to fix that in code

Reply:
Fixed.